dask/dataframe/shuffle.py

```diff
@@ -1051,8 +1032,8 @@ def _compute_partition_stats(
     maxes = column.map_partitions(M.max, meta=column)
     lens = column.map_partitions(len, meta=column)
     mins, maxes, lens = compute(mins, maxes, lens, **kwargs)
-    mins = remove_nans(mins)
-    maxes = remove_nans(maxes)
+    mins = pd.Series(mins).fillna(method="bfill").tolist()
+    maxes = pd.Series(maxes).fillna(method="bfill").tolist()
```

Comment:
I've not looked deeply, but do we need these `pd.Series(...)` constructors? Should they not already always be series? 

```suggestion
    mins = mins.fillna(method="bfill").tolist()
    maxes = maxes.fillna(method="bfill").tolist()
```